client/post-editor/editor-drawer/taxonomies.jsx

```diff
@@ -34,26 +35,29 @@ function EditorDrawerTaxonomies( { siteId, postType, postTerms, taxonomies } ) {
 						<Accordion
 							key={ taxonomy.name }
 							title={ taxonomy.label }
-							icon={ <Gridicon icon="category" /> }
+							icon={ <Gridicon icon="folder" /> }
 						>
-							Work in Progress
-						</Accordion>
-					);
-				} else {
-					return (
-						<Accordion
-							key={ taxonomy.name }
-							title={ taxonomy.label }
-							icon={ <Gridicon icon="tag" /> }
-						>
-							<TermTokenField
-								postTerms={ postTerms }
+							<TermSelector
+								selectedTermIds={ postTerms ? map( postTerms[ taxonomy.name ], 'ID' ) : [] }
 								taxonomyName={ taxonomy.name }
-								taxonomyLabel={ taxonomy.label }
 							/>
 						</Accordion>
 					);
 				}
+
+				return (
+					<Accordion
```

Comment:
Since the render behavior is mostly consistent between the taxonomy types, aside from the content of the accordion and the Gridicon `icon`, maybe we could consolidate this a bit more?

```diff
iff --git a/client/post-editor/editor-drawer/taxonomies.jsx b/client/post-editor/editor-drawer/taxonomies.jsx
index 70a219e..4973ff7 100644
--- a/client/post-editor/editor-drawer/taxonomies.jsx
+++ b/client/post-editor/editor-drawer/taxonomies.jsx
@@ -25,37 +25,25 @@ function EditorDrawerTaxonomies( { siteId, postType, postTerms, taxonomies } ) {
                                <QueryTaxonomies { ...{ siteId, postType } } />
                        ) }
                        { map( taxonomies, ( taxonomy ) => {
-                               if ( 'post_format' === taxonomy.name ) {
+                               const { name, label, hierarchical } = taxonomy;
+                               if ( 'post_format' === name ) {
                                        // Post format has its own dedicated accordion
                                        return;
                                }
 
-                               if ( taxonomy.hierarchical ) {
-                                       return (
-                                               <Accordion
-                                                       key={ taxonomy.name }
-                                                       title={ taxonomy.label }
-                                                       icon={ <Gridicon icon="folder" /> }
-                                               >
-                                                       <TermSelector
-                                                               selectedTermIds={ postTerms ? map( postTerms[ taxonomy.name ], 'ID' ) : [] }
-                                                               taxonomyName={ taxonomy.name }
-                                                       />
-                                               </Accordion>
-                                       );
-                               }
-
                                return (
                                        <Accordion
-                                               key={ taxonomy.name }
-                                               title={ taxonomy.label }
-                                               icon={ <Gridicon icon="tag" /> }
-                                       >
-                                               <TermTokenField
-                                                       postTerms={ postTerms }
-                                                       taxonomyName={ taxonomy.name }
-                                                       taxonomyLabel={ taxonomy.label }
-                                               />
+                                               key={ name }
+                                               title={ label }
+                                               icon={ <Gridicon icon={ hierarchical ? 'folder' : 'tag' } /> }>
+                                               { hierarchical
+                                                       ? <TermSelector
+                                                               selectedTermIds={ postTerms ? map( postTerms[ taxonomy.name ], 'ID' ) : [] }
+                                                               taxonomyName={ taxonomy.name } />
+                                                       : <TermTokenField
+                                                               postTerms={ postTerms }
+                                                               taxonomyName={ taxonomy.name }
+                                                               taxonomyLabel={ taxonomy.label } /> }
                                        </Accordion>
                                );
                        } ).filter( Boolean ) }
```